modules/tensor_mechanics/src/materials/ComputeAxisymmetric1DSmallStrain.C

```diff
@@ -24,15 +24,34 @@ validParams<ComputeAxisymmetric1DSmallStrain>()
 ComputeAxisymmetric1DSmallStrain::ComputeAxisymmetric1DSmallStrain(
     const InputParameters & parameters)
   : Compute1DSmallStrain(parameters),
-    _scalar_out_of_plane_strain_coupled(isCoupledScalar("scalar_out_of_plane_strain")),
-    _scalar_out_of_plane_strain(_scalar_out_of_plane_strain_coupled
-                                    ? coupledScalarValue("scalar_out_of_plane_strain")
-                                    : _zero),
-    _out_of_plane_strain_coupled(isCoupled("out_of_plane_strain")),
-    _out_of_plane_strain(_out_of_plane_strain_coupled ? coupledValue("out_of_plane_strain") : _zero)
+    _subblock_id_provider(isParamValid("subblock_index_provider")
+                              ? &getUserObject<SubblockIndexProvider>("subblock_index_provider")
+                              : nullptr),
+    _has_out_of_plane_strain(isParamValid("out_of_plane_strain")),
+    _out_of_plane_strain(_has_out_of_plane_strain ? coupledValue("out_of_plane_strain") : _zero),
+    _has_scalar_out_of_plane_strain(isParamValid("scalar_out_of_plane_strain")),
+    _nscalar_strains(
+        _has_scalar_out_of_plane_strain ? coupledScalarComponents("scalar_out_of_plane_strain") : 0)
 {
-  if (_out_of_plane_strain_coupled && _scalar_out_of_plane_strain_coupled)
+  if (_has_out_of_plane_strain && _has_scalar_out_of_plane_strain)
     mooseError("Must define only one of out_of_plane_strain or scalar_out_of_plane_strain");
+
+  if (!_has_out_of_plane_strain && !_has_scalar_out_of_plane_strain)
+    mooseError("Must define either out_of_plane_strain or scalar_out_of_plane_strain");
+
+  // in case when the provided scalar_out_of_plane_strain is not a coupled
+  // scalar variable, still set _nscalar_strains = 1 but return its default value 0
+  if (coupledScalarComponents("scalar_out_of_plane_strain") == 0)
+    _nscalar_strains = 1;
+
+  if (_has_scalar_out_of_plane_strain)
+  {
+    _scalar_out_of_plane_strain.resize(_nscalar_strains);
+    for (unsigned int i = 0; i < _nscalar_strains; ++i)
+    {
```

Comment:
Sorry, one last minor issue:  remove curly braces around one-line block.  Apparently clang-format doesn't check for this...